docs/tutorial/build_system.md

```diff
@@ -1,25 +1,220 @@
+      - target/snitch_cluster/generated/snax_alu/snax_alu_csrman_CsrManager.sv
+      - target/snitch_cluster/generated/snax_alu/snax_alu_streamer_StreamerTop.sv
+      # Level 1
+      - hw/snax_alu/src/snax_alu_shell_wrapper.sv
+      - target/snitch_cluster/generated/snax_alu/snax_alu_csrman_wrapper.sv
+      - target/snitch_cluster/generated/snax_alu/snax_alu_streamer_wrapper.sv
+      # Level 2
+      - target/snitch_cluster/generated/snax_alu/snax_alu_wrapper.sv
+```
+
+The `-target` parameter is a switch to include the files listed below it. Here, we use `snax-alu` as the target. Take note of this as we will use this definition in the makefile build later.
+
+The files are listed according to a hierarchical structure from the base modules up to the system modules. This means that Level 0 contains modules that are not dependent on any other modules. Level 1 is dependent on Level 0 modules. Likewise, Level 2 is dependent on both Level 1 and Level 0 modules.
+
+At Level 0, you will notice that we declared the `snax_alu_csr.sv` and `snax_alu_pe.sv` which are the base modules of our main SNAX ALU accelerator. It is important to declare the generated files which are the `snax_alu_csrman_CsrManager.sv` and `snax_alu_streamer_StreamerTop.sv`.
+
+At Level 1, the `snax_alu_shell_wrapper.sv` is dependent on the ALU CSR and PEs and hence it is listed above the two. The wrappers for the CSR manager and streamer are also dependent on the Chisel-generated modules. Finally, Level 2 is the accelerator wrapper which is directly connected to the cluster system.
+
+When you make your own accelerator you need to declare all the RTL files in this format too. Make sure it's under the SNAX Accelerator section just to make sure you don't mess up the structure.
+
+### What About the Naming of the Generated Files?
+
+The names of the generated files are dependent on the configuration file you set. If you recall in the `snax_acc_cfg`, there is a `snax_acc_name` parameter: 
+
+```hjson
+snax_acc_cfg: {
+    snax_acc_name: "snax_alu"
+    snax_tcdm_ports: 16,
+    snax_num_rw_csr: 3,
+    snax_num_ro_csr: 2,
+    snax_streamer_cfg: {$ref: "#/snax_alu_streamer_template" }
+},
+```
+This is prefixed at the beginning of each generated file. Moreover, it also sets the generated directory for each accelerator you add. Such that we have:
+
+```yml
+- target: snax-alu
+    files:
+      # Level 0
+      # ... Add some other files here
+      - target/snitch_cluster/generated/${snax_acc_name}/${snax_acc_name}_csrman_CsrManager.sv
+      - target/snitch_cluster/generated/${snax_acc_name}/${snax_acc_name}_streamer_StreamerTop.sv
+      # Level 1
+      # ... Add some other files here
+      - target/snitch_cluster/generated/${snax_acc_name}/${snax_acc_name}_streamer_wrapper.sv
+      # Level 2
+      - target/snitch_cluster/generated/${snax_acc_name}/${snax_acc_name}_wrapper.sv
+```
+
+Where each `${snax_acc_name}` is substituted with the parameter name you specified.
```

Comment:
Are these to be substituted by the human? Otherwise I would put:
```suggestion
Where you should substitute `${snax_acc_name}`  with the parameter name you specified.
```

Reply:
Yes, they are human substituted 😅